### PR TITLE
Pipe 24072

### PR DESCRIPTION
**Pull request by shivamnegi94**
**Title:** [Component/Feature] Short Description of the Change

**Summary:**
Provide a brief overview of what the PR does and why it is needed.

**Details:**
Explain the changes in detail, including any relevant context or background information.

**Related Issues:**
Reference any related issues or tickets 

**Testing Instructions:**
Describe how the changes were tested. Include any test cases or steps to verify the functionality.
Please include the below test scenario with your changes.
- Create the resource and execute terraform apply. (Verify the resource is created)
- Execute terraform apply again without any changes. (Verify no changes should be done)
- Update the resource and execute terraform apply. (Verify the resource updated successfully)
- Remove the content from resource file and execute terraform apply. (Verify the resource has been deleted)
- Add again the resource and execute the terraform apply. (Verify the resource is created)
- Verify the import the resource file is working fine.
- In case of remote entity, Verify for both default and non default branch.

**Screenshots:**
Include before and after screenshots to visually demonstrate the changes.
**Checklist:**
- [x] Code changes are well-documented.
- [x] Tests have been added/updated.
- [x] Changes have been tested locally.
- [ ] Documentation has been updated.

<details>
  <summary>PR Check triggers</summary>
  
- Build: `trigger build`
- Sub Category Field Check: `trigger subcategoryfieldcheck`
- gitleaks: `trigger gitleaks`

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -6,7 +6,7 @@ require (
 	github.com/antihax/optional v1.0.0
 	github.com/aws/aws-sdk-go v1.46.4
 	github.com/docker/docker v24.0.5+incompatible
-	github.com/harness/harness-go-sdk v0.4.21
+	github.com/harness/harness-go-sdk v0.4.23
```
Comment (shashankJainHarness) [inline go.mod:9]: this update is intentional?

File: internal/service/pipeline/pipeline_filters/resource_pipeline_filters.go [hunk 2]
```diff
@@ -499,6 +533,12 @@ func readPipelineFilter(d *schema.ResourceData, filter *nextgen.PipelineFilter)
 			if attr, ok := hCdProperties["deploymentTypes"]; ok {
 				cdProperties["deployment_types"] = attr.(string)
 			}
+			if attr, ok := hCdProperties["serviceDefinitionTypes"].([]interface{}); ok {
+				cdProperties["service_definition_types"] = attr[0]
+			}
+			if attr, ok := hCdProperties["serviceDefinitionTypes"].([]interface{}); ok {
```
Comment (shashankJainHarness) [inline internal/service/pipeline/pipeline_filters/resource_pipeline_filters.go:539]: this looks duplicated, please check

File: go.mod (hunk 1 above)
Comment (shivamnegi94) [reply go.mod:9]: Yes, we have changes in harness-go-sdk

File: internal/service/pipeline/pipeline_filters/resource_pipeline_filters.go (hunk 2 above)
Comment (shivamnegi94) [reply internal/service/pipeline/pipeline_filters/resource_pipeline_filters.go]: resolved the issue

**Review by shashankJainHarness — APPROVED** [review]

**Pull request merged**